src/app/static/src/app/components/adr/SelectRelease.vue

```diff
@@ -0,0 +1,167 @@
+            </span>
+            <br />
+        </div>
+        <div>
+            <input
+                type="radio"
+                id="useRelease"
+                value="useRelease"
+                v-model="choiceADR"
+            />
+            <label
+                for="useRelease"
+                v-translate="'useRelease'"
+                class="pr-1"
+            ></label>
+            <span class="icon-small" v-tooltip="translate('releaseTooltip')">
+                <help-circle-icon></help-circle-icon>
+            </span>
+            <br />
+        </div>
+
+        <label
+            for="releaseSelector"
+            class="font-weight-bold"
+            v-translate="'releases'"
+        ></label>
+        <tree-select
+            id="releaseSelector"
+            :multiple="false"
+            :searchable="true"
+            :options="releaseOptions"
+            :placeholder="translate('select')"
+            :disabled="!useRelease"
+            v-model="newReleaseId"
+        >
+            <label
+                slot="option-label"
+                slot-scope="{ node }"
+                v-html="node.raw.customLabel"
+            >
+            </label>
+        </tree-select>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import { mapActionByName, mapStateProp, mapMutationByName } from "../../utils";
+    import TreeSelect from "@riophae/vue-treeselect";
+    import { ADRState } from "../../store/adr/adr";
+    import { HelpCircleIcon } from "vue-feather-icons";
+    import { VTooltip } from "v-tooltip";
+    import i18next from "i18next";
+    import { Language } from "../../store/translations/locales";
+    import { RootState } from "../../root";
+    import {ADRMutation} from "../../store/adr/mutations";
+
+    interface Data {
+        newReleaseId: string | null;
+        choiceADR: "useData" | "useRelease";
+    }
+
+    interface Methods {
+        getReleases: (id: string) => void;
+        clearReleases: () => void;
+        translate(text: string): string;
+    }
+
+    interface Computed {
+        releases: any[];
+        selectedDatasetOrRelease: string | null;
+        releaseOptions: any[];
+        useRelease: boolean;
+        currentLanguage: Language;
+    }
+
+    interface Props {
+        newDatasetId: string | null;
+    }
+
+    const namespace = "adr";
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        components: {
+            TreeSelect,
+            HelpCircleIcon,
+        },
+        props: {
+            newDatasetId: String,
+        },
+        data() {
+            return {
+                newReleaseId: null,
+                choiceADR: "useData",
+            };
+        },
+        computed: {
+            releases: mapStateProp<ADRState, any[]>(
+                namespace,
+                (state: ADRState) => state.releases
+            ),
+            selectedDatasetOrRelease() {
```

Comment:
This computed value is just used to determine if the selections are in a valid state isn't it? I don't really like bringing in `newDatasetId` into it - it seems a but irrelevant if it's just going to be used to generate a boolean. Would this be a little more explicable as something like 
```
valid() {
  return (this.choiceADR === "useData) || this.newReleaseId;
}
```
That could fit in nicely with Marks' suggestion of adding a `valid` event